notebooks/working/ho_deconfounder_on_sim_data.py

```diff
@@ -0,0 +1,512 @@
+#
+# We estimate three models:
+# - Model 1: A multinomial logit with the correct original specification, EXCEPT we ommit the travel distance variable in the specification without trying to adjust for it. 
+# - Model 2: We use the deconfounder algorithm to try to recover the confounder (travel distance). In this method, we use all the variables in each mode's utility to recover that mode's confounder.
+# - Model 3: We use the deconfounder algorithm to try to recover the confounder (travel distance), but this time, we only use travel time and cost in the factor model, instead of all the variables in the utility specification of each mode. 
+#
+# We compare the estimates of the coefficients on travel time and cost to the true estimates used in the simulation. The main findings of this exercise are the following:
+# - Using the true variable believed to be confounded (i.e. method 3 where only travel time and cost are used to recover the confounder) leads to a better recovery of the true confounder. This suggests that it may be better to run the deconfounder algorithm based on a hypothesized causal graph, rather than just running it on all the observed covariates. 
+# - Similar to what we found in the investigating_decounfounder notebook, the effectiveness of the deconfounder algorithm is very sensitive to small deviations in the recovered confounder. Although method 3 returns a relatively good fit of the true confounder, the adjusted coefficients on travel time and cost do not improve, and the coefficients on the recovered confounder are highly insignificant. This raises questions about the usefulness of the deconfounder algorithm in practice.
+
+# ## Useful function
+
+def specifications(mnl_specification, num_modes):
+    newDict= dict()
+
+    for i in range(1,num_modes+1):
+        variables = []
+        # Iterate over all the items in dictionary and filter items which has even keys
+        for (key, value) in mnl_specification.items():
+           # Check if key is even then add pair to new dictionary
+    
+
+
+
+            if any(isinstance(sub, list) for sub in value):
+                for sub in value:
+                    if isinstance(sub, list):
+                        if (i in sub): 
+                            variables.append(key)
+                            
+                    else:
+                        if i == sub:
+                            variables.append(key)
+
+            else:
+                if i in value:
+        #             print(variables)
+
+                    variables.append(key)
+
+        newDict[i] = variables
+
+    return newDict
+
+# ## Analysis
+
+PATH = '../../data/raw/'
+file_name = 'simulated_long_format_bike_data.csv'
+
+data = pd.read_csv(PATH+file_name)
+data = data.drop('Unnamed: 0', axis = 1)
+data.columns
+
+# +
+X_columns = ['total_travel_time',
```

Comment:
Can you collect all the constants (e.g., `PATH`, `'mode_id'`), lists of strings (e.g. `X_columns`), and notebook parameters in general into one cell at the top of the notebook? The idea is to make it easy to use tools like papermill to programmatically execute the notebooks in a workflow.